lib/datadog/appsec.rb

```diff
@@ -8,6 +8,16 @@ module Datadog
   module AppSec
     include Configuration
 
+    class << self
+      def components
+        Datadog.send(:components)
+      end
+
+      def processor
+        components.appsec.processor if components.appsec
+      end
```

Comment:
Thanks for the detailed explanation.

Yes, I had `components.appsec&.processor` before, but CO for old rubies 💣  so I had to change it. 

I'll hoist the component in a local variable

Reply:
Feel free to update the "OLD RUBY PAIN TRACKER" spreadsheet linked on the internal Ruby channel with this instance, as it's an example of something where you lost time going back and forth (and me commenting on it) because we're still supporting those Rubies.